src/rocksdb_options.rs

```diff
@@ -23,37 +23,91 @@ use merge_operator::{self, MergeOperatorCallback, MergeOperands, full_merge_call
               partial_merge_callback};
 use comparator::{self, ComparatorCallback, compare_callback};
 
-#[derive(Copy, Clone)]
-pub struct RocksDBOptions {
+pub struct BlockBasedOptions {
+    inner: rocksdb_ffi::RocksDBBlockBasedTableOptions,
+}
+
+pub struct Options {
     pub inner: rocksdb_ffi::RocksDBOptions,
-    block_options: rocksdb_ffi::RocksDBBlockBasedTableOptions,
 }
 
-impl RocksDBOptions {
-    pub fn new() -> RocksDBOptions {
+impl Drop for Options {
+    fn drop(&mut self) {
+        unsafe {
+            rocksdb_ffi::rocksdb_options_destroy(self.inner);
+        }
+    }
+}
+
+impl Drop for BlockBasedOptions {
+    fn drop(&mut self) {
+        unsafe {
+            rocksdb_ffi::rocksdb_block_based_options_destroy(self.inner);
+        }
+    }
+}
+
+impl BlockBasedOptions {
+    pub fn new() -> BlockBasedOptions {
+        let block_opts = unsafe {rocksdb_ffi::rocksdb_block_based_options_create() };
+        let rocksdb_ffi::RocksDBBlockBasedTableOptions(opt_ptr) = block_opts;
+        if opt_ptr.is_null() {
+            panic!("Could not create rocksdb block based options".to_string());
+        }
+        BlockBasedOptions{ inner: block_opts, }
+    }
+
+    pub fn set_block_size(&mut self, size: u64) {
+        unsafe {
+            rocksdb_ffi::rocksdb_block_based_options_set_block_size(
+                self.inner, size);
+        }
+    }
+
+    //TODO figure out how to create these in a Rusty way
+    ////pub fn set_filter(&mut self, filter: rocksdb_ffi::RocksDBFilterPolicy) {
+    ////    unsafe {
+    ////        rocksdb_ffi::rocksdb_block_based_options_set_filter_policy(
+    ////            self.inner, filter);
+    ////    }
+    ////}
+
+    ////pub fn set_cache(&mut self, cache: rocksdb_ffi::RocksDBCache) {
+    ////    unsafe {
+    ////        rocksdb_ffi::rocksdb_block_based_options_set_block_cache(
+    ////            self.inner, cache);
+    ////    }
+    ////}
+
+    ////pub fn set_cache_compressed(&mut self, cache: rocksdb_ffi::RocksDBCache) {
+    ////    unsafe {
+    ////        rocksdb_ffi::rocksdb_block_based_options_set_block_cache_compressed(
+    ////            self.inner, cache);
+    ////    }
+    ////}
```

Comment:
I agree that these hack methods should be removed from the API until a nicer implementation arises.